docs/proposals/multi-template-partitionting-and-scheduler-extensibility/README.md

```diff
@@ -0,0 +1,647 @@
+
+To implement this capability, we can allow users to select custom strategies in `PropagationPolicy`.
+This way, the Scheduler will know which `AssignReplica` plugin's scheduling results to choose.
+
+```go
+type ReplicaSchedulingStrategy struct {
+    // ReplicaSchedulingType determines how the replicas is scheduled when karmada propagating
+    // a resource. Valid options are Duplicated and Divided.
+    // "Duplicated" duplicates the same replicas to each candidate member cluster from resource.
+    // "Divided" divides replicas into parts according to number of valid candidate member
+    // clusters, and exact replicas for each cluster are determined by ReplicaDivisionPreference.
+    // +kubebuilder:validation:Enum=Duplicated;Divided
+    // +kubebuilder:default=Divided
+    // +optional
+    ReplicaSchedulingType ReplicaSchedulingType `json:"replicaSchedulingType,omitempty"`
+
+    // ReplicaDivisionPreference determines how the replicas is divided
+    // when ReplicaSchedulingType is "Divided". Valid options are Aggregated and Weighted.
+    // "Aggregated" divides replicas into clusters as few as possible,
+    // while respecting clusters' resource availabilities during the division.
+    // "Weighted" divides replicas by weight according to WeightPreference.
+    // +kubebuilder:validation:Enum=Aggregated;Weighted
+    // +optional
+    ReplicaDivisionPreference ReplicaDivisionPreference `json:"replicaDivisionPreference,omitempty"`
+
+    // WeightPreference describes weight for each cluster or for each group of cluster
+    // If ReplicaDivisionPreference is set to "Weighted", and WeightPreference is not set, scheduler will weight all clusters the same.
+    // +optional
+    WeightPreference *ClusterPreferences `json:"weightPreference,omitempty"`
+	
+    // 🌟 New 🌟
+    // +optional
+	CustomSchedulingStrategy string `json:"customSchedulingStrategy,omitempty"`
+}
+```
+
+<!--
+CustomSchedulingStrategy 字段允许用户直接配置自定义策略的名称，此字段为可选，如果设置为 `default` 则与未设置同义。
+它通过 PropagationPolicy 告诉调度器在调度过程中如何选择 AssignReplica 阶段的插件和该插件对应的策略，我们认为单个插件是有可能对应多个自定义策略的。
+比如在我们预想的 `gang` 插件中，它应当支持 `gang-divided` 和 `gang-no-divided` 两个策略。
+
+在这里要注意的是，当用户配置 CustomReplicaSchedulingOptions 后，可能会导致调度器忽略用户配置的其他调度设置（如ReplicaSchedulingType）。
+是否使用这些参数将会由自定义插件来决定。比如 A 插件它基于 Divided Aggregated 策略实现了一些其他的能力，那么它就需要使用用户配置的其他内容，
+再比如 B 插件它实现了一个完全不同之前的调度策略，那么用户配置的内容很有可能是毫无意义的。
+-->
+
+The `CustomSchedulingStrategy` field allows users to directly configure the name of a custom strategy.
+This field is optional; if set to `default`, it is synonymous with being unset.  
+It informs the scheduler via the `PropagationPolicy`
+on how to select the `AssignReplica` plugin and its corresponding strategy during the scheduling process.
+We believe that a single plugin could correspond to multiple custom strategies.  
+For example, in our envisioned `gang` plugin, it should support both `gang-divided` and `gang-no-divided` strategies.
```

Comment:
Form API perspective, all we need to declare are the expectations, not the underlayer implementation. I mean assign replia plugin might be one of the options to implement.

Reply:
Agreed, users do not need to care about the underlying implementation or understand the assignReplica plugin when using the API. They only need to indicate their desired strategy